### PR TITLE
feat(analyze-dashboard): display collaborators list in popover

### PR DESCRIPTION
**Pull request by andrewazores**
Display the list of space collaborators in a popover component upon click of the collaborator count
label when viewing the Analyze Dashboard for any given space

related to https://openshift.io/openshiftio/Openshift_io/plan/detail/847

~~Still TODO: add search widget in popover~~

image

image

### CODE REVIEW

**Comment by joshuawilson** [comment]
@fabric8-ui/uxd please review.
@fabric8-ui/docs this is something new.

**Comment by andrewazores** [comment]
@aptmac ah good catch. Will update momentarily.

**Comment by andrewazores** [comment]
Actually - I can make the count label accurate momentarily. Making the popover list contain the actual entire correct list will take a little longer since it'll mean implementing some infinite scroll there.

**Comment by andrewazores** [comment]
There's a problem with using infinite scroll in the popover list however. The returned results are not sorted alphabetically by default and the `CollaboratorService` does not expose a way to query with a sort order, so the popover list will have to display the results simply as published by the `CollaboratorService` paging. Sorting can then be added to the `CollaboratorService` and/or the `-wit` backend to properly enable sorting the popover list.

**Comment by andrewazores** [comment]
Marking WIP again as resolving the sorting issue above will take some time.

**Comment by andrewazores** [comment]
image

image

image

Latest update removes the fixed list component and uses an infinite scroll instead (styled to look the same), populated by CollaboratorService pagination. The search widget is removed temporarily since this requires CollaboratorService/WIT work to work correctly.

**Comment by joshuawilson** [comment]
I think it does make sense.
Can you please write up the task on planner to get the support?

File: src/app/dashboard-widgets/edit-space-description-widget/edit-space-description-widget.component.less [hunk 1]
```diff
@@ -173,3 +173,35 @@
...
+.collaborator-list-scroll {
+  max-height: inherit;
+  overflow-x: hidden;
+  overflow-y: auto;
+}
+
+.collaborator-item {
+  padding-top: 6px;
+  padding-right: 6px;
+  padding-bottom: 12px;
```
[the remark this answers is not in the document]
Comment (andrewazores) [reply src/app/dashboard-widgets/edit-space-description-widget/edit-space-description-widget.component.less]: Looks like I forgot to finish fiddling with the padding there, whoops. I've updated again. Top and bottom were meant to have one thin padding size, and left and right were meant to have another thicker padding size, just to try to match the design mock.

**Comment by Preeticp** [comment]
Instructions and screenshots for this feature have been updated through: https://github.com/fabric8io/fabric8-online-docs/pull/453

**Pull request merged**